libraries/rating-bar/src/main/java/com/trendyol/uicomponents/ratingbar/RatingBarView.kt

```diff
@@ -0,0 +1,90 @@
+package com.trendyol.uicomponents.ratingbar
+
+import android.content.Context
+import android.graphics.Color
+import android.util.AttributeSet
+import android.widget.LinearLayout
+import androidx.annotation.ColorInt
+import com.trendyol.uicomponents.ratingbar.databinding.ViewRatingBarBinding
+
+class RatingBarView @JvmOverloads constructor(
+    context: Context, attrs: AttributeSet? = null, defStyleAttr: Int = 0
+) : LinearLayout(context, attrs, defStyleAttr) {
+
+    private var starCount: Int = 0
+    @ColorInt
+    private var starHighlightColor: Int = Color.YELLOW
+    @ColorInt
+    private var starDefaultColor: Int = Color.GRAY
+
+    private val binding: ViewRatingBarBinding = inflate(R.layout.view_rating_bar)
+
+    init {
+        context.theme?.obtainStyledAttributes(
+            attrs,
+            R.styleable.RatingBarView,
+            defStyleAttr,
+            0
+        )?.apply {
+            try {
+                starCount = getInteger(R.styleable.RatingBarView_starCount, 0)
+                starHighlightColor =
+                    getColor(
+                        R.styleable.RatingBarView_starHighlightColor,
+                        context.color(R.color.star_highlight)
+                    )
+                starDefaultColor = getColor(
+                    R.styleable.RatingBarView_starDefaultColor,
+                    context.color(R.color.star_default)
+                )
+
+                setViewState(createViewState())
+            } finally {
+                recycle()
+            }
+        }
+    }
+
+    /**
+     *
+     * Sets star count to highlight with provided @param starHighlightColor
+     *
+     * @param starCount: starCount to highlight
+     */
+    fun setStarCount(starCount: Int) {
+        this.starCount = starCount
```

Comment:
If the maximum star count is predefined like "5", the function should throw the exception or log the message

Reply:
As I said before, we will make these in later releases.